analyses/somalia/scripts/som_foodinsec_trigger.py

```diff
@@ -0,0 +1,69 @@
+warnings.filterwarnings('ignore')
+
+from pathlib import Path
+import os
+import sys
+path_mod = f"{Path(os.path.dirname(os.path.realpath(__file__))).parents[2]}/"
+sys.path.append(path_mod)
+from indicators.food_insecurity.config import Config
+from indicators.food_insecurity.ipc_definemetrics import define_trigger_percentage, define_trigger_increase
+
+admin_level=0
+country="somalia"
+#suffix of filenames
+suffix=""
+config=Config()
+parameters = config.parameters(country)
+country_folder = os.path.join(config.DIR_PATH, config.ANALYSES_DIR, country)
+fewsnet_dir = os.path.join(country_folder, config.DATA_DIR, config.FEWSWORLDPOP_PROCESSED_DIR)
+fewsnet_filename = config.FEWSWORLDPOP_PROCESSED_FILENAME.format(country=country,admin_level=admin_level,suffix=suffix)
+globalipc_dir=os.path.join(country_folder,config.DATA_DIR, config.GLOBALIPC_PROCESSED_DIR)
+globalipc_path=os.path.join(globalipc_dir,f"{country}_globalipc_admin{admin_level}{suffix}.csv")
+
+adm_bound_path= os.path.join(country_folder,config.DATA_DIR,config.SHAPEFILE_DIR,parameters[f"path_admin{admin_level}_shp"])
+
+df_fn=pd.read_csv(os.path.join(fewsnet_dir,fewsnet_filename))
+df_fn["source"]="FewsNet"
+df_gipc=pd.read_csv(globalipc_path)
+df_gipc["source"]="GlobalIPC"
+
+df=pd.concat([df_fn,df_gipc])
+df=df.replace(0,np.nan)
+df["date"]=pd.to_datetime(df["date"])
+df["year"]=df["date"].dt.year
+df["month"]=df["date"].dt.month
+df["trigger_ML1_4_20"]=df.apply(lambda x: define_trigger_percentage(x,"ML1",4,20),axis=1)
+df["trigger_ML1_3_30"]=df.apply(lambda x: define_trigger_percentage(x,"ML1",3,30),axis=1)
+df["trigger_ML1_3_5i"]=df.apply(lambda x: define_trigger_increase(x,"ML1",3,5),axis=1)
+df[f"threshold_reached_ML1"]=np.where((df[f"trigger_ML1_4_20"]==1) | ((df[f"trigger_ML1_3_30"]==1) & (df[f"trigger_ML1_3_5i"]==1)),1,0)
+
+df["trigger_ML2_4_20"]=df.apply(lambda x: define_trigger_percentage(x,"ML2",4,20),axis=1)
+df["trigger_ML2_3_30"]=df.apply(lambda x: define_trigger_percentage(x,"ML2",3,30),axis=1)
+df["trigger_ML2_3_5i"]=df.apply(lambda x: define_trigger_increase(x,"ML2",3,5),axis=1)
+df[f"threshold_reached_ML2"]=np.where((df[f"trigger_ML2_4_20"]==1) | ((df[f"trigger_ML2_3_30"]==1) & (df[f"trigger_ML2_3_5i"]==1)),1,0)
+
+df["trigger_ML1_3_20"] = df.apply(lambda x: define_trigger_percentage(x,"ML1",3,20),axis=1)
+df["trigger_ML1_3_5in"] = df.apply(lambda x: define_trigger_increase(x,"ML1",3,5),axis=1)
+df["trigger_ML1_4_2half"] = df.apply(lambda x: define_trigger_percentage(x,"ML1",4,2.5),axis=1)
+
+df["trigger_ML2_3_20"] = df.apply(lambda x: define_trigger_percentage(x,"ML2",3,20),axis=1)
+df["trigger_ML2_3_5in"] = df.apply(lambda x: define_trigger_increase(x,"ML2",3,5),axis=1)
+df["trigger_ML2_4_2half"] = df.apply(lambda x: define_trigger_percentage(x,"ML2",4,2.5),axis=1)
+
+# determine whether national trigger is met
+df['threshold_reached_ML1'] =  np.where((df['trigger_ML1_3_20']==1) & ((df['trigger_ML1_3_5in'] )==1 | (df['trigger_ML1_4_2half'] == 1)), 1, 0)
```

Comment:
The projected national population in Phase 3 and above exceed 20%, AND
(The national population in Phase 3 is projected to increase by 5 percentage points, OR
The projected national population in Phase 4 or above is 2.5%)